### PR TITLE
feat: Add generic restrictedStray behavior

### PR DESCRIPTION
**Pull request by AndyTechGuy**
Add a new kind of stray behavior, `restrictedStray`. This behavior is similar to the stray behavior, except it restricts the stray area to a rectangle are defined in a `StrayRestrictionComponent`. This is useful for cases where a NPC character must stay within the area of a particular building, or inside the area of a city.

### CODE REVIEW

File: src/main/java/org/terasology/behaviors/actions/NearbyBlockRestricted.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import org.terasology.logic.behavior.core.BehaviorState;
+import org.terasology.math.geom.Rect2i;
+import org.terasology.minion.move.MinionMoveComponent;
+
+import java.util.List;
+import java.util.Random;
+import org.terasology.navgraph.WalkableBlock;
+
+/**
+ * Sets a marketkeeper character's {@link MinionMoveComponent} target to a random nearby block inside the area of
```
Comment (skaldarnar) [inline src/main/java/org/terasology/behaviors/actions/NearbyBlockRestricted.java]: JavaDoc is still specific for the MR shopkeeper NPC 😉

Comment (AndyTechGuy) [reply src/main/java/org/terasology/behaviors/actions/NearbyBlockRestricted.java]: whoops! The javadoc here has been edited to be more generic

File: src/main/java/org/terasology/behaviors/actions/NearbyBlockRestricted.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        allowedRegion = strayRestriction.allowedRegion;
+    }
+
+    @Override
+    public BehaviorState modify(Actor actor, BehaviorState result) {
+        if (random.nextInt(100) > (99 - moveProbability)) {
+            MinionMoveComponent moveComponent = actor.getComponent(MinionMoveComponent.class);
+            if (moveComponent.currentBlock != null) {
+                WalkableBlock target = findRandomNearbyBlockInMarket(moveComponent.currentBlock);
+                moveComponent.target = target.getBlockPosition().toVector3f();
```
Comment (agent-q1) [inline src/main/java/org/terasology/behaviors/actions/NearbyBlockRestricted.java]: Maybe change function name to 
`findRandomNearbyBlockInAllowedRegion` or `findRandomNearbyBlockInRestrictedRegion`

**Review by agent-q1 — COMMENTED** [review]
Tested it out, works well. Still think it's too much unnecessary computation, definitely makes it robust though. A small change in naming suggested

File: src/main/java/org/terasology/behaviors/actions/NearbyBlockRestricted.java (hunk 2 above)
Comment (AndyTechGuy) [reply src/main/java/org/terasology/behaviors/actions/NearbyBlockRestricted.java]: Whoops, missed this one while changing the documentation to remove all market references, thank you for pointing it out! Fixed 👍

**Review by skaldarnar — APPROVED** [review]

**Comment by skaldarnar** [comment]
Same tests failing as always - we need to fix this 🙈

**Pull request merged**